packages/react/src/animation/TextRevealAnimation/TextRevealAnimation.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useEffect} from 'react'
+import clsx from 'clsx'
+import {useTextRevealAnimationLines} from './useTextRevealAnimationLines'
+
+import styles from './TextRevealAnimation.module.css'
+import {BaseProps} from '../../component-helpers'
+
+export type TextRevealAnimationProps = BaseProps<HTMLSpanElement> &
+  Omit<React.HTMLAttributes<HTMLSpanElement>, 'children'> & {
+    children: React.ReactChild
+  }
+
+export function TextRevealAnimation({children, ...rest}: TextRevealAnimationProps) {
+  const [animationStarted, setAnimationStarted] = React.useState(false)
+  const [hasHeroWipeAnimated, setHasHeroWipeAnimated] = React.useState(false)
```

Comment:
Should we rename these to be component-agnostic? For example, `hasAnimated` and `setHasAnimated`.

Reply:
🤦 good spot, i forgot to rename this after porting it from article template... will fix 👍